BatteryExamine.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+
+public class BatteryExamine
+{
+    private readonly IBatteryLimits batteryLimits;
+    public BatteryExamine(IBatteryLimits iBatteryLimits)
+    {
+        this.batteryLimits = iBatteryLimits;
+    }
+    public bool BatteryIsOk(float batteryTemperature, float stateOfCharge, float chargeRate)
+    {
+        bool checkTemperatureResult = CompareTemperatureWithRange(batteryTemperature);
+        bool checkStateOfChargeResult = CompareStateOfChargeWithRange(stateOfCharge);
+        bool checkChargeRateResult = CompareChargeRateWithRange(chargeRate);
+        return (checkTemperatureResult && checkStateOfChargeResult && checkChargeRateResult);
+    }
+    private bool CompareTemperatureWithRange(float batteryTemperature)
+    {
+        if (batteryTemperature < batteryLimits.minTemperature || batteryTemperature > batteryLimits.maxTemperature)
+        {
+            CompareTemperatureWithLimits(batteryTemperature);
+            DisplayMessage("Temperature");
+            return false;
+        }
+        return true;
+    }
+    public bool CompareStateOfChargeWithRange(float stateOfCharge)
+    {
+        if (stateOfCharge < batteryLimits.minStateOfCharge || stateOfCharge > batteryLimits.maxStateOfCharge)
+        {
+            CompareStateOfChargeWithLimits(stateOfCharge);
+            DisplayMessage("State of Charge");
+            return false;
+        }
+        return true;
+    }
+    public bool CompareChargeRateWithRange(float chargeRate)
+    {
+        if (chargeRate > batteryLimits.minChargeRate)
+        {
+            DisplayMaximumLimit("Charge Rate", batteryLimits.minChargeRate);
+            DisplayMessage("Charge Rate");
+            return false;
+        }
+        return true;
+    }
+    public void CompareTemperatureWithLimits(float batteryTemperature)
+    {
+        if(batteryTemperature < batteryLimits.minTemperature)
+        {
+            DisplayMinimumLimit("Temperature", batteryLimits.minTemperature);
+        }
+        else
+        {
+            DisplayMaximumLimit("Temperature", batteryLimits.maxTemperature);
+        }
+    }
+    public void CompareStateOfChargeWithLimits(float stateOfCharge)
+    {
+        if (stateOfCharge < batteryLimits.minStateOfCharge)
+        {
+            DisplayMinimumLimit("State Of Charge", batteryLimits.minStateOfCharge);
+        }
+        else
+        {
+            DisplayMaximumLimit("State Of Charge", batteryLimits.maxStateOfCharge);
+        }
+    }
+   
+    static void DisplayMessage(string batteryParameter)
+    {
+        Console.WriteLine("{0} is out of range!", batteryParameter);
+    }
+    static void DisplayMinimumLimit(string batteryParameter,float batteryParameterLimit)
```

Comment:
How DisplayMinimumLimit and DisplayMaximumLimit differ from implementation (Duplicate Code)

Reply:
Removed duplicate implementation